spring-social-zotero/src/main/java/org/springframework/social/zotero/api/impl/GroupsTemplate.java

```diff
@@ -350,4 +412,36 @@ public void deleteItem(String groupId, String citationKey, Long citationVersion)
         }
 
     }
+    
+    private ZoteroUpdateItemsStatuses getStatusesFromResponse(List<ItemCreationResponse> responses,
+            List<String> itemsKeys) {
+        // Zotero is not sending failed items keys in response. So, we have to compute
+        // failed items keys on our own using success and unchanged keys
+
+        ZoteroUpdateItemsStatuses statuses = new ZoteroUpdateItemsStatuses();
+        List<String> successKeys = new ArrayList<>();
+        List<String> unchangedKeys = new ArrayList<>();
+        Set<String> failedMessages = new HashSet<String>();
+        for (ItemCreationResponse response : responses) {
+            successKeys.addAll(extractItemKeys(response.getSuccess(), e -> e.getValue()));
+            unchangedKeys.addAll(extractItemKeys(response.getUnchanged(), e -> e.getValue()));
+            failedMessages.addAll(response.getFailed().entrySet().stream().map(e -> e.getValue().getMessage())
```

Comment:
the code is more important than the message. The code should indicate what the issue was, the message would be just an explanation. We might want to have our own message or if we ever internationalize the app, we don't want to be stuck with the messages.